packages/v3-sdk/src/entities/pool.ts

```diff
@@ -195,15 +195,12 @@ export class Pool {
     const zeroForOne = outputAmount.currency.equals(this.token1)
 
     const {
-      amountSpecifiedRemaining,
       amountCalculated: inputAmount,
       sqrtRatioX96,
       liquidity,
       tickCurrent,
     } = await this.swap(zeroForOne, outputAmount.quotient * NEGATIVE_ONE, sqrtPriceLimitX96)
 
-    invariant(amountSpecifiedRemaining === ZERO, 'INSUFICIENT_LIQUIDITY')
```

Comment:
It's by passing the sdk checking for incomplete ticks?
do you think we can have a flag on pool initial to bypass it?

Reply:
maybe the suggestion is not worth to add complexity into the pool creation

Reply:
actually this logic is added by me previously to throw if the liquidity is not enough for the whole swap. Actually v3 does support partial swap so it's better to remove the logic I added before

Reply:
it happens when swap takes all the liquidity and the tick goes to max/min tick. And the current liquidity becomes 0 even though there're still valid ticks